annotation-utils/src/main/java/edu/isi/vista/annotationutils/AceCorpusTextSource.java

```diff
@@ -0,0 +1,52 @@
+package edu.isi.vista.annotationutils;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+import edu.isi.nlp.symbols.Symbol;
+import com.google.common.base.Optional;
+import edu.isi.nlp.io.OriginalTextSource;
+import java.io.File;
+import java.nio.file.Path;
+import java.io.IOException;
+
+
+public final class AceCorpusTextSource implements OriginalTextSource{
+    private  File aceDataDirectory;
+    private File[] sourceFileDirectories;
+
+    public Optional<String> getOriginalText(final Symbol docID) throws IOException {
+        String rawSourceText = "";

```

Comment:
Avoid using empty strings to represent anything except an actual empty string.  Here I think you are using it to represent "we couldn't find the source raw text", so you want the type of `rawSourceText` to be `Optional<String>`, initialized to `Optional.absent()`.  Then set it to `Optional.of(....)` if you find it.